OneSignalSDK/onesignal/core/src/main/java/com/onesignal/core/internal/operations/impl/OperationRepo.kt

```diff
@@ -347,13 +359,31 @@ internal class OperationRepo(
 
     internal fun getNextOps(bucketFilter: Int): List<OperationQueueItem>? {
         return synchronized(queue) {
-            val startingOp =
-                queue.firstOrNull {
-                    it.operation.canStartExecute &&
-                        _newRecordState.canAccess(it.operation.applyToRecordId) &&
-                        it.bucket <= bucketFilter
+            var startingOp: OperationQueueItem? = null
+            // Search for the first operation that is qualified to execute
+            for (queueItem in queue) {
+                val operation = queueItem.operation
+
+                // Ensure the operation is in an executable state
+                if (!operation.canStartExecute ||
+                    !_newRecordState.canAccess(
+                        operation.applyToRecordId,
+                    ) || queueItem.bucket > bucketFilter
+                ) {
+                    continue
+                }
+
+                // Ensure the operation does not have empty JWT if identity verification is on
+                if (_configModelStore.model.useIdentityVerification &&
+                    _identityModelStore.model.jwtToken.isNullOrEmpty()
+                ) {
+                    continue
                 }
```

Comment:
Yes, it only need to be checked once per method call because it does not rely on any operation specific reference. I moved it out of the loop and put back the code `queue.firstOrNull {...}` in the fixup commit.